File: Difficulty: Easy/Modify the Array/modify-the-array.py
#{ 
 # Driver Code Starts
#Initial Template for Python 3

# } Driver Code Ends
#User function Template for python3

class Solution:
    def modifyAndRearrangeArr(self, arr):
        n = len(arr)
        
        # Step 1: Modify the array by merging adjacent equal non-zero values
        for i in range(n - 1):
            if arr[i] != 0 and arr[i] == arr[i + 1]:
                arr[i] *= 2
                arr[i + 1] = 0
        
        # Step 2: Move all non-zero values to the front, and zeros to the end
        result = []
        for num in arr:
            if num != 0:
                result.append(num)
        
        # Fill the remaining spaces with 0s
        result += [0] * (n - len(result))
        
        return result



#{ 
 # Driver Code Starts.
#Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    while t > 0:
        arr = list(map(int, input().split()))
        ob = Solution()
        ans = ob.modifyAndRearrangeArr(arr)
        print(*ans)
        t -= 1


# } Driver Code Ends

File: Difficulty: Easy/Modify the Array/modify-the-array.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
    vector<int> modifyAndRearrangeArray(vector<int> &arr) {
       int n = arr.size();
       for(int i =0;i<n-1;i++){
           if(arr[i]==arr[i+1] and arr[i]!=0){
               arr[i]*=2;
               arr[i+1]=0;
               i++;
           }
       }
       int i=0;
       for(int j = 0;j<n; j++){
           if(arr[j]){
               while(i<j and arr[i]!=0) i++;
               if(i<j)swap(arr[i],arr[j]);
           }
       }
       return arr;
    }
};


//{ Driver Code Starts.

int main() {
    string ts;
    getline(cin, ts);
    int t = stoi(ts);
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        Solution obj;
        vector<int> ans = obj.modifyAndRearrangeArray(arr);
        for (int i = 0; i < ans.size(); i++) {
            cout << ans[i] << " ";
        }
        cout << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends